app/src/main/java/nerd/tuxmobil/fahrplan/congress/settings/SettingsFragment.kt

```diff
@@ -4,44 +4,64 @@ import android.annotation.TargetApi
 import android.app.Activity.RESULT_OK
 import android.content.Context
 import android.content.Intent
+import android.media.RingtoneManager
+import android.net.Uri
 import android.os.Build
 import android.os.Bundle
-import android.preference.Preference
-import android.preference.Preference.OnPreferenceChangeListener
-import android.preference.Preference.OnPreferenceClickListener
-import android.preference.PreferenceCategory
-import android.preference.PreferenceFragment
-import android.preference.PreferenceScreen
 import android.provider.Settings
 import androidx.core.os.bundleOf
+import androidx.preference.EditTextPreference
+import androidx.preference.Preference
+import androidx.preference.Preference.OnPreferenceChangeListener
+import androidx.preference.Preference.OnPreferenceClickListener
+import androidx.preference.PreferenceCategory
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
 import nerd.tuxmobil.fahrplan.congress.BuildConfig
 import nerd.tuxmobil.fahrplan.congress.R
 import nerd.tuxmobil.fahrplan.congress.alarms.AlarmServices
 import nerd.tuxmobil.fahrplan.congress.contract.BundleKeys
 import nerd.tuxmobil.fahrplan.congress.extensions.getAlarmManager
 import nerd.tuxmobil.fahrplan.congress.extensions.toSpanned
+import nerd.tuxmobil.fahrplan.congress.preferences.AlarmTonePreference
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
 import nerd.tuxmobil.fahrplan.congress.utils.FahrplanMisc
 
-class SettingsFragment : PreferenceFragment() {
+class SettingsFragment : PreferenceFragmentCompat() {
+
+    private companion object {
+
+        const val REQUEST_CODE_ALARM_TONE = 3439
+
+    }
+
+    private lateinit var appRepository: AppRepository
```

Comment:
Why not this?

```kotlin
private val appRepository = AppRepository
```

Reply:
:heavy_check_mark: Changed in 23b10ca4822a133ed18915842987e132b4a9205a.